### PR TITLE
feat: Implement left navigation level 1 for sites - EXO-65477 - Meeds-io/MIPs#72

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 1]
```diff
@@ -37,10 +38,16 @@
...
+import org.exoplatform.portal.config.UserACL;
+import org.exoplatform.portal.config.UserPortalConfig;
 import org.exoplatform.portal.config.UserPortalConfigService;
 import org.exoplatform.portal.config.model.PortalConfig;
+import org.exoplatform.portal.mop.SiteFilter;
 import org.exoplatform.portal.mop.SiteKey;
 import org.exoplatform.portal.mop.SiteType;
+import org.exoplatform.portal.mop.navigation.Scope;
+import org.exoplatform.portal.mop.service.LayoutService;
+import org.exoplatform.portal.mop.user.*;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]: need to import explicitly the needed classes

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 2]
```diff
@@ -1688,4 +1706,92 @@ public static RelationshipManager getRelationshipManager() {
     }
     return relationshipManager;
   }
+
+  public static SiteRestEntity toSiteRestEntity(PortalConfig site, HttpServletRequest request, SiteFilter siteFilter) {
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
  private static SiteRestEntity buildSiteEntity(PortalConfig site, HttpServletRequest request, SiteFilter siteFilter) {
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 3]
```diff
@@ -1688,4 +1706,92 @@ public static RelationshipManager getRelationshipManager() {
...
+            site.getDescription(),
+            accessPermissions,
+            editPermission,
+            isDefaultSite(site.getName()) || site.isDisplayed(),
+            site.getDisplayOrder(),
+            isDefaultSite(site.getName()),
+            rootNode == null ? null : toUserNodeRestEntity(rootNode.getChildren(), true, getOrganizationService(), getLayoutService(), getUserACL()));
+
+  }
+  public static List<SiteRestEntity> toSiteRestEntities(List<PortalConfig> sites,
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
  public static List<SiteRestEntity> buildSiteEntities(List<PortalConfig> sites,
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 4]
```diff
@@ -1688,4 +1706,92 @@ public static RelationshipManager getRelationshipManager() {
...
+        UserPortal userPortal = userPortalCfg.getUserPortal();
+        UserNavigation navigation = userPortal.getNavigation(new SiteKey(siteType.getName(), site.getName()));
+        rootNode = userPortal.getNode(navigation, Scope.ALL, UserNodeFilterConfig.builder().build(), null);
+      } catch (Exception e) {
+        LOG.error("Error while getting site {} navigations for user {}", site.getName(), currentUser, e);
+      }
+    }
+    return new SiteRestEntity(siteType,
+            site.getName(),
+            !org.apache.commons.lang.StringUtils.isBlank(displayName) ? displayName : site.getName(),
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]: add this import org.apache.commons.lang.StringUtils explicitly with the list of imports

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 5]
```diff
@@ -1688,4 +1706,92 @@ public static RelationshipManager getRelationshipManager() {
...
+    }
+    return userACL;
+  }
+  private static LayoutService getLayoutService() {
+    if (layoutService == null) {
+      layoutService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(LayoutService.class);
+    }
+    return layoutService;
+  }
+  public static boolean isDefaultSite(String siteName) {
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
  private static boolean isDefaultSite(String siteName) {
```

File: component/service/src/main/java/org/exoplatform/social/rest/entity/SiteRestEntity.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/entity/SiteRestEntity.java]: Rename to SiteEntity instead of SiteRestEntity

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**